ios/Video/RCTVideo.m

```diff
@@ -1522,106 +1566,351 @@ - (void)removeFromSuperview
+  return [self loadingRequestHandling:loadingRequest];
+}
+
+- (void)resourceLoader:(AVAssetResourceLoader *)resourceLoader
+didCancelLoadingRequest:(AVAssetResourceLoadingRequest *)loadingRequest {
+  NSLog(@"didCancelLoadingRequest");
+}
+
+- (BOOL)loadingRequestHandling:(AVAssetResourceLoadingRequest *)loadingRequest {
+  if (self->_requestingCertificate) {
+    return YES;
+  } else if (self->_requestingCertificateErrored) {
+    return NO;
+  }
+  _loadingRequest = loadingRequest;
+  NSURL *url = loadingRequest.request.URL;
+  NSString *contentId = url.host;
+  if (self->_drm != nil) {
+    NSString *contentIdOverride = (NSString *)[self->_drm objectForKey:@"contentId"];
+    if (contentIdOverride != nil) {
+      contentId = contentIdOverride;
+    }
+    NSString *drmType = (NSString *)[self->_drm objectForKey:@"type"];
+    if ([drmType isEqualToString:@"fairplay"]) {
+      NSString *certificateStringUrl = (NSString *)[self->_drm objectForKey:@"certificateUrl"];
+      if (certificateStringUrl != nil) {
+        NSURL *certificateURL = [NSURL URLWithString:[certificateStringUrl stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+          NSData *certificateData = [NSData dataWithContentsOfURL:certificateURL];
+          if ([self->_drm objectForKey:@"base64Certificate"]) {
+            certificateData = [[NSData alloc] initWithBase64EncodedData:certificateData options:NSDataBase64DecodingIgnoreUnknownCharacters];
+          }
+          
+          if (certificateData != nil) {
+            NSData *contentIdData = [contentId dataUsingEncoding:NSUTF8StringEncoding];
+            AVAssetResourceLoadingDataRequest *dataRequest = [loadingRequest dataRequest];
+            if (dataRequest != nil) {
+              NSError *spcError = nil;
+              NSData *spcData = [loadingRequest streamingContentKeyRequestDataForApp:certificateData contentIdentifier:contentIdData options:nil error:&spcError];
+              // Request CKC to the server
+              NSString *licenseServer = (NSString *)[self->_drm objectForKey:@"licenseServer"];
+              if (spcError != nil) {
+                [self finishLoadingWithError:spcError];
+                self->_requestingCertificateErrored = YES;
+              }
+              if (spcData != nil) {
+                if(self.onGetLicense) {
+                  NSString *spcStr = [[NSString alloc] initWithData:spcData encoding:NSASCIIStringEncoding];
+                  self->_requestingCertificate = YES;
+                  self.onGetLicense(@{@"spc": spcStr,
+                                      @"contentId": contentId,
+                                      @"spcBase64": [[[NSData alloc] initWithBase64EncodedData:certificateData options:NSDataBase64DecodingIgnoreUnknownCharacters] base64EncodedStringWithOptions:0],
```

Comment:
Use `spcData` instead of `certificateData`, but it's not working.

`@"spcBase64": [spcData base64EncodedStringWithOptions:0]` is working for me. 

P.S I'm not objective-c developer.

Reply:
thank you!